compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2355,6 +2359,61 @@ public void visit(BLangTupleDestructure tupleDeStmt, AnalyzerData data) {
         if (type.tag != TypeTags.SEMANTIC_ERROR) {
             checkTupleVarRefEquivalency(tupleDeStmt.pos, tupleDeStmt.varRef,
                                         tupleDeStmt.expr.getBType(), tupleDeStmt.expr.pos, data);
+            if (tupleDeStmt.expr.getKind() == LIST_CONSTRUCTOR_EXPR) {
+                analyzeDestructuringWithListBindingPattern(tupleDeStmt.varRef,
+                        (BLangListConstructorExpr) tupleDeStmt.expr);
+            }
+        }
+    }
+
+    // TODO: This is temporary. Remove this once #42352 gets fixed
+    private void analyzeDestructuringWithListBindingPattern(BLangTupleVarRef tupleVarRef,
```

Comment:
No errors logged for something like

```
[{name: n2}, {name: n3}, {name: n4}] = [...[{name: "", id: 0}, {name: "", id: 0}, {name: "", id: 0}]];
```

which was previously also an error.

Reply:
Now we support this, hence fixed